Lib/test/test_unittest/testmock/testmock.py

```diff
@@ -1652,6 +1652,15 @@ def test_mock_unsafe(self):
         m.aseert_foo_call()
         m.assrt_foo_call()
 
+    # gh-100739
+    def test_mock_safe_with_spec(self):
+        class Foo(object):
+            def assert_bar(self):
```

Comment:
I would suggest to also test a name like `assertSome`. Because `_` might be important.